foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/oxm/XMLUnmarshaller.java

```diff
@@ -969,4 +1012,20 @@ public final void setDisableSecureProcessing(boolean disableSecureProcessing) {
         platformUnmarshaller.setDisableSecureProcessing(disableSecureProcessing);
     }
 
+    public Boolean isLogPayload() {
+        return logPayload;
+    }
+
+    public void setLogPayload(Boolean logPayload) {
+        this.logPayload = logPayload;
+    }
+
+    private void logUnmarshall(Object object) {
```

Comment:
wouldn't it be better to have this within the implementation of 'this.platformUnmarshaller.unmarshal' method?